src/wifi_hal_priv.h

```diff
@@ -780,6 +780,55 @@ INT wifi_hal_setRMBeaconRequest(UINT apIndex, mac_address_t peer_mac, wifi_Beaco
 INT wifi_hal_cancelRMBeaconRequest(UINT apIndex, UCHAR dialogToken);
 INT wifi_hal_configNeighborReports(UINT apIndex, bool enable, bool auto_resp);
 INT wifi_hal_setNeighborReports(UINT apIndex, UINT numNeighborReports, wifi_NeighborReport_t *neighborReports);
+
+/**
+* @brief Add a vendor-specific Information Element to the AP's beacon/probe response
+*
+* @note Certain drivers may override or not allow additional IEs to be added causing 
+*    the "Failed to set beacon parameters" error message to be logged from the 
+*    `ieee802_11_set_beacon` / "UPDATE_BEACON" function.
+*
+* @param[in]  apIndex  Access point index
+* @param[in]  oui      Pointer to the 3-byte Organization Unique Identifier
+* @param[in]  data     Pointer to the vendor-specific data to be included in the IE
+* @param[in]  data_len Length of the vendor-specific data in bytes
+*
+* @return The status of the operation
+* @retval WIFI_HAL_SUCCESS if successful
+* @retval WIFI_HAL_INVALID_ARGUMENTS if any of the arguments are invalid
+* @retval WIFI_HAL_INTERNAL_ERROR if any internal error is detected
+* @retval WIFI_HAL_UNSUPPORTED if the driver does not support adding vendor-specific IEs
+*
+* @execution Synchronous
+* @sideeffect Updates beacon and probe response frames
+*
+*/
+INT wifi_addVendorSpecificIE(INT apIndex, const UCHAR *oui, UCHAR *data, UINT data_len);
+
+/**
+* @brief Remove a **single instance** of a vendor-specific Information Element from the AP's beacon/probe response (if present)
+* 
+* @note Certain drivers may override or not allow additional IEs to be removed causing
+*    the "Failed to set beacon parameters" error message to be logged from the
+*    `ieee802_11_set_beacon` / "UPDATE_BEACON" function.
+*
+* @param[in]  apIndex  Access point index
+* @param[in]  oui      Pointer to the 3-byte Organization Unique Identifier
+* @param[in]  data     Pointer to the vendor-specific data to match for removal
+* @param[in]  data_len Length of the vendor-specific data in bytes
+*
+* @return The status of the operation
+* @retval WIFI_HAL_SUCCESS if removed successfully or if the IE was not present
+* @retval WIFI_HAL_INVALID_ARGUMENTS if any of the arguments are invalid
+* @retval WIFI_HAL_INTERNAL_ERROR if any internal error is detected
+* @retval WIFI_HAL_UNSUPPORTED if the driver does not support removing vendor-specific IEs
+*
+* @execution Synchronous
+* @sideeffect Updates beacon and probe response frames
+*
+*/
+INT wifi_removeVendorSpecificIE(INT apIndex, const UCHAR *oui, UCHAR *data, UINT data_len);
```

Comment:
Same comment as above. If you are planning to use this function in OneWifi, the declaration of this function should not be here. It should be in the halinterface appropriate include file. 

wifi_hal_priv.h should only have declaration of functions used internally within rdk-wifi-hal project